drivers/timer/nrf_rtc_timer.c

```diff
@@ -120,12 +122,22 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 
 	k_spinlock_key_t key = k_spin_lock(&lock);
 	u32_t cyc, dt, t = counter();
+	u32_t unannounced = counter_sub(t, last_count);
 	bool zli_fixup = IS_ENABLED(CONFIG_ZERO_LATENCY_IRQS);
 
+	/* If we haven't announced for more than half the 24-bit wrap
+	 * duration, then force an announce to avoid loss of a wrap
+	 * event.  This can happen if new timeouts keep being set
+	 * before the existing one triggers the interrupt.
+	 */
+	if (unannounced >= COUNTER_HALF_SPAN) {
+		ticks = 0;
```

Comment:
@pabigot I wonder if we could, just set the IRQ pending via NVIC here

Reply:
I guess we could.  I don't see what we gain, though.  We know the code works correctly for a delay of zero ticks, and probably that's exactly the path it will take in the following code.

If we do it here, then maybe we should do it for an incoming request for zero ticks too, then maybe we should have a special exit path past all the clock setting stuff, then we'd have to consider whether it's ok to leave the compare register at its old setting, then....

Could we make that something to consider for 2.2?

Reply:
Yes